cpp/benchmarks/flow_simulation_ode_seir.cpp

```diff
@@ -32,61 +32,92 @@ namespace mio
+
+                const double Nj_inv = 1.0 / (pop[Sj] + pop[Ej] + pop[Ij] + pop[Rj]);
+                const double coeffStoE =
+                    params.template get<ContactPatterns<ScalarType>>().get_cont_freq_mat().get_matrix_at(t)(i.get(),
+                                                                                                            j.get()) *
+                    params.template get<TransmissionProbabilityOnContact<ScalarType>>()[i] * Nj_inv;
+
+                double dummy_S = y[Si] * y[Ij] * coeffStoE;
+                dydt[Si] -= dummy_S;
+                dydt[Ei] += dummy_S;
+            }
+
+            dydt[Ii] += (1.0 / params.get<TimeExposed<ScalarType>>()[i]) * y[Ei];
+            dydt[Ii] -= (1.0 / params.get<TimeInfected<ScalarType>>()[i]) * y[Ii];
+            dydt[Ri] = (1.0 / params.get<TimeInfected<ScalarType>>()[i]) * y[Ii];
+        }
     }
 };
 
 template <class Model>
 void setup_model(Model& model)
 {
-    double total_population                                                                            = 10000;
-    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Exposed)}]   = 100;
-    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Infected)}]  = 100;
-    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Recovered)}] = 100;
-    model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Susceptible)}] =
-        total_population -
-        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Exposed)}] -
-        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Infected)}] -
-        model.populations[{mio::Index<mio::oseir::InfectionState>(mio::oseir::InfectionState::Recovered)}];
-    // suscetible now set with every other update
-    // params.nb_sus_t0   = params.nb_total_t0 - params.nb_exp_t0 - params.nb_inf_t0 - params.nb_rec_t0;
-    model.parameters.template set<mio::oseir::TimeExposed>(5.2);
-    model.parameters.template set<mio::oseir::TimeInfected>(6);
-    model.parameters.template set<mio::oseir::TransmissionProbabilityOnContact>(0.04);
-    model.parameters.template get<mio::oseir::ContactPatterns>().get_baseline()(0, 0) = 10;
+    double total_population = 10000;
+    const auto num_groups   = model.parameters.get_num_groups();
+    for (AgeGroup i = 0; i < num_groups; i++) {
+        model.populations[{i, oseir::InfectionState::Exposed}]   = 100 / static_cast<double>((size_t)num_groups);
+        model.populations[{i, oseir::InfectionState::Infected}]  = 100 / static_cast<double>((size_t)num_groups);
+        model.populations[{i, oseir::InfectionState::Recovered}] = 100 / static_cast<double>((size_t)num_groups);
+        model.populations[{i, oseir::InfectionState::Susceptible}] =
+            total_population / static_cast<double>((size_t)num_groups) -
```

Comment:
```suggestion
        model.populations[{i, oseir::InfectionState::Exposed}]   = 100.0 / static_cast<size_t>(num_groups);
        model.populations[{i, oseir::InfectionState::Infected}]  = 100.0 / static_cast<size_t>(num_groups);
        model.populations[{i, oseir::InfectionState::Recovered}] = 100.0 / static_cast<size_t>(num_groups);
        model.populations[{i, oseir::InfectionState::Susceptible}] =
            total_population / static_cast<size_t>(num_groups) -
```
I think this is nicer, since it avoids mixed casts